litert/vendors/qualcomm/core/builders/hadamard_transform_op_builder.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Qualcomm Innovation Center, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "litert/vendors/qualcomm/core/builders/hadamard_transform_op_builder.h"
+
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+#include <variant>
+#include <cmath>
+#include "QnnOpDef.h"  // from @qairt
+#include "absl/numeric/bits.h"
+#include "litert/vendors/qualcomm/core/builders/op_builder.h"
+#include "litert/vendors/qualcomm/core/utils/log.h"
+#include "litert/vendors/qualcomm/core/wrappers/op_wrapper.h"
+#include "litert/vendors/qualcomm/core/wrappers/tensor_wrapper.h"
+namespace qnn {
+
+namespace {
+constexpr std::size_t kInputIndex = 0;
+constexpr std::size_t kWeightIndex = 1;
+constexpr std::size_t kOutputIndex = 0;
+
+std::optional<float> GetSylvesterHadamardScale(const TensorWrapper& weight) {
+  if (weight.GetRank() != 2) return std::nullopt;
+  if (weight.GetDim(0) != weight.GetDim(1)) return std::nullopt;
+  const std::uint32_t n = weight.GetDim(0);
+
+  // Ensure the size is power-of-two.
+  if (n == 0 || (n & (n - 1) != 0)) return std::nullopt;
```

Comment:
Please be aware of the operator precedence.
```suggestion
  if (n == 0 || (n & (n - 1)) != 0) return std::nullopt;
```